public/app/features/explore/utils/decorators.ts

```diff
@@ -77,6 +79,22 @@ export const decorateWithFrameTypeMetadata = (data: PanelData): ExplorePanelData
   };
 };
 
+export const decorateWithCorrelations = ({
+  queries,
+  correlations,
+}: {
+  queries: DataQuery[] | undefined;
+  correlations: CorrelationData[] | undefined;
+}) => {
+  return (data: PanelData): PanelData => {
+    if (!isEmpty(queries) && !isEmpty(correlations)) {
+      const queryRefIdToDataSourceUid = mapValues(groupBy(queries, 'refId'), '0.datasource.uid');
+      attachCorrelationsToDataFrames(data.series, correlations!, queryRefIdToDataSourceUid);
```

Comment:
i think in this case not using `isEmpty` may be better as we then don't have to do a non-null assertion on `correlations`
```suggestion
    if (queries?.length && correlations?.length) {
      const queryRefIdToDataSourceUid = mapValues(groupBy(queries, 'refId'), '0.datasource.uid');
      attachCorrelationsToDataFrames(data.series, correlations, queryRefIdToDataSourceUid);
```

Reply:
Fixed in d4ac82b4f240271dab7a3490827bdf59d079ee90